File: LaboRTI2023_sources/Data/DataBaseException.cpp
#include "DataBaseException.h"

const int DataBaseException::QUERY_ERROR = 1;
const int DataBaseException::EMPTY_RESULT_SET = 2;

DataBaseException::DataBaseException() : Exception() { }

DataBaseException::DataBaseException(string msg, int code) : Exception(msg)
{
    this->setCode(code);
}

void DataBaseException::setCode(int code)
{
    if(code == QUERY_ERROR || code == EMPTY_RESULT_SET)
    {
        this->code = code;
    }
}

int DataBaseException::getCode() { return this->code; }

File: LeMaraicher_ClientAchat/src/Modele/Exceptions/DataBaseException.java
package Modele.Exceptions;

public class DataBaseException extends Exception{

    public static final int QUERY_ERROR = 1;
    public static final int EMPTY_RESULT_SET = 2;
    private int code;

    public DataBaseException() {
        super();
    }
    public DataBaseException(String msg, int code) {
        super(msg);
        this.code = code;
    }

    int getCode() { return code; }
}
